### DISCUSSION TITLE
Multiple characters at once!

### DISCUSSION THREAD

**Discussion by ReflectionRip** [oobabooga/text-generation-webui#3525] [Ideas]
Anyone looking into the idea of having more than once character running at once?

I think it would be fun to have a 1 on 2 or 1 on 3 chat. Or possibly a web based chat room where 1 or more people in the room are AI characters.

**Comment by mattieshadow** [comment]
You can definitely just make the character sheet two people, and have the AI play as both at the same time no problem. Just give it a basic format to follow or something in the character sheet for speaking as both.

**Reply by ReflectionRip** [reply]
That never works out well. The two characters end up blending into one.

**Reply by mattieshadow** [reply]
Perhaps a model issue. The model I use for roleplaying does an excellent job at just this. I would recommend MythosMix, you can find it on TheBloke's hugging face account, if you care.

**Comment by missionfloyd** [comment]
[Sillytavern](https://github.com/SillyTavern/SillyTavern) can do it.

**Reply by ReflectionRip** [reply]
I'll give this a try. Thanks.

**Comment by TechScorpion** [comment]
Would it be difficult to change this Text Generation Webgui to allow to talk to more than one character at once (group chats)? If it could be done, could it be an option please?

Even though Sillytavern offers that option that app seems more controversial so it would be good if there was a basic group chat option in this app that we could use instead, that was more professional and not controversial.

**Reply by FemBoxbrawl** [reply]
facts, if it were in ooba it would be way easier

**Reply by Woisek** [reply]
What do you mean by "this app is controversial"?

**Reply by TechScorpion** [reply]
I meant SillyTavern (not this oobabooba app). Though it might have just seemed that way from the characters in the documentation/some screenshots/youtube vid I'd seen, it has a 'Waifu' mode and the FAQ says:

> Tavern is an app to facilitate these roleplaying chats:
> ...
>     In the background, it does the needful to prepare the AI for your character roleplay. Specifically, it will send a system prompt (instructions for the AI) that primes the AI to follow certain rules that make for a good chat session. **Optionally, it can also try to allow the roleplay to go into an "adult" direction.** 

I think it could still be done in oobabooga text gen in a different way, that could be more serious or useful (even though it could be just fun too) as well as quicker/easier (without having to install other big apps). eg. there could be a prompt where it lets the characters know what the group chat is about/where and about what the group might discuss (without the stuff from SillyTavern giving the option to make it into an "adult" roleplay).

**Reply by Woisek** [reply]
First, this "Waifu"-Mode is just a display/layout option and is mainly for the anime dudes out there, as you can clearly see from the screenshots in the FAQs. If you use it or not, it has nothing to do with what is generated while chatting.
I personally create and rp with "real" like characters, so I never needed that in any way.

Second, what you took out of the FAQ is basically something, that every chat AI does. The only difference is, with ST, one is not censored and so it can also be used for adult chatting. Comparing to the commercial versions online, this is not a "problem", this is the freedom for everyone to do as he likes.
Plus, it has absolutely nothing to do if you do a chat with a single bot, or a group chat.
If you are concerned, that your chat drifts into "naughtiness", then just make sure that:

- you use/create cards without any sexual hints
- you don't use "horny" models
- you insert SFW settings into ST

If you were familiar to how this all works, you would know that already, and not state that ST is "controversial", because you can configure it the one or the other way.

**Reply by TechScorpion** [reply]
I was just giving my opinion of it at the time. Maybe I should have said "in my opinion.." at the start.
But like I said, for the reasons I gave above, I still think the group chat option would be useful in this oobabooga text gen app though (quicker, easier, less to set up/run.). Like I said, I'd based what I said on what I'd seen in the documentation and screencaps and youtube vid, not from using ST at the time.

It might also help having group chat option in oobabooga if you ever needed to give the sources of your info if it came out with something useful (maybe by also having them discuss attached documents). It would probably be a more serious source having it be oobabooga text gen than sillytavern (since people could see ST had the other stuff that some other people might not think is that ok and also doesn't look quite as serious/as a professional app - not a game like one).

**Reply by Woisek** [reply]
I can understand, that a group chat functionality would be great in ooba and there speaks nothing against it. But what you have addressed is something that ST had always had, this wasn't something "new" or special, it always worked that way. Also, you judge the program by its look, and this is also not right to do. I, too, used ooba for a long time and did rp, but found out, that ST fits me more in how I can use and interact with it. Yes, ST is more trimmed towards role playing, but that doesn't mean that you can't do "serious" stuff with it. You just have to configure it that way, like any other program, too. In the end, everything is just a tool you have to configure and use how you want it.
And it would be definitely great if ooba includes technical problematic group chats. Maybe they come up with a better solution on to handle this.

**Reply by spekcool** [reply]
Hey, I also think that ST is annoying, just like the author of the latest comments, Woisek. (Bro, if you like ST, then use it. I DON'T like ST, and the reasons don't matter. So, I'm in favor of OOga having its own implementation of a similar idea, rather than relying on awkward workarounds.)

**Comment by mykeehu** [comment]
I opened an issue here, maybe it will happen...
https://github.com/oobabooga/text-generation-webui/issues/6540
